package/src/custom/tests/School/unit-php/phpunit.xml.dist

```diff
@@ -0,0 +1,50 @@
+<!--
+/*
+ * Your installation or use of this SugarCRM file is subject to the applicable
+ * terms available at
+ * http://support.sugarcrm.com/Resources/Master_Subscription_Agreements/.
+ * If you do not agree to all of the applicable terms or do not have the
+ * authority to bind the entity as an authorized representative, then do not
+ * install or use this SugarCRM file.
+ *
+ * Copyright (C) SugarCRM Inc. All rights reserved.
+ */
+-->
+
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.1/phpunit.xsd"
+         backupGlobals="false"
+         backupStaticAttributes="false"
+         colors="true"
+         convertErrorsToExceptions="true"
+         convertNoticesToExceptions="true"
+         convertWarningsToExceptions="true"
+         processIsolation="false"
+         stopOnFailure="false"
+         bootstrap="../../../../tests/unit-php/TestBootstrap.php"
+         beStrictAboutOutputDuringTests="true"
+         beStrictAboutTestsThatDoNotTestAnything="true"
+         beStrictAboutTestSize="true"
+         beStrictAboutCoversAnnotation="false"
+>
+
+    <php>
+        <ini name="intl.default_locale" value="en"/>
+        <ini name="intl.error_level" value="0"/>
+        <ini name="memory_limit" value="-1"/>
+    </php>
+    <testsuites>
+        <testsuite name="all">
+            <directory>./</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist processUncoveredFilesFromWhitelist="false">
+            <directory suffix=".php">../../</directory>
```

Comment:
At this point, I think these are the only files you want to generate code coverage for. You could continue to add to `custom/gradebook_fake`. You could also make changes to `custom/Extension/modules/Schedulers/Ext/ScheduledTasks/AddStudentToGradebookJob.php` and/or `custom/modules/Contacts/Students_Gradebook.php`. So I think you should have this as your whitelist:

```
<directory suffix=".php">../../custom/gradebook_fake</directory>
<file>../../custom/Extension/modules/Schedulers/Ext/ScheduledTasks/AddStudentToGradebookJob.php</file>
<file>../../custom/modules/Contacts/Students_Gradebook.php</file>
```

That would only generate code coverage for your code. I don't think you can avoid having to maintain this list. It's just going to have to be a part of your DoD. If you add a new class to `custom/modules/Accounts`, then you'll need to add that on its own. But if you add another class to `custom/gradebook_fake`, then you won't need to add it because it'll already be captured by the `<directory>` directive.

Obviously, this is untested, so the only way to know if it works is to run it. You can either run it selectively from the command line or always run it.

To run it on the command line, you would run your tests with `/path/to/phpunit --coverage-html /path/to/output/file`. You just have to decide what type of coverage report you want to run (and the location where you want to write the file). The choices are:

```
--coverage-html /tmp/report
--coverage-clover /tmp/coverage.xml
--coverage-php /tmp/coverage.serialized
--coverage-text
> /tmp/logfile.txt
--log-junit /tmp/logfile.xml
--testdox-html /tmp/testdox.html
--testdox-text /tmp/testdox.txt
```

To run it always, it's just another addition to your config file. Again, it's just a matter of choosing the type and the path to the file for output.

```
<logging>
    <log type="coverage-html" target="/tmp/report"/>
</logging>
```

Reply:
Thanks for the info on how to run the code coverage!  I was able to get it working.  I tweaked the settings and ended up with the following:

```
        <whitelist addUncoveredFilesFromWhitelist="false" processUncoveredFilesFromWhitelist="false">
            <directory suffix=".php">../../..</directory>
            <exclude>
                <directory suffix=".php">../..</directory>
            </exclude>
        </whitelist>
```

This sets the coverage to run on the custom directory and excludes the custom/tests directory.  The downside to this is that if someone fails to test a file at all, it won't show up in the code coverage report.  However, I feel more confident that we'll remember to check if a PR has appropriate testing than if the PR updates the phpunit.xml.dist.  We could set this file and forget about it.  What do you think?